include/zephyr/bluetooth/conn.h

```diff
@@ -196,6 +196,81 @@ struct bt_conn_le_data_len_param {
 	BT_CONN_LE_DATA_LEN_PARAM(BT_GAP_DATA_LEN_MAX, \
 				  BT_GAP_DATA_TIME_MAX)
 
+/** Connection subrating parameters for LE connections */
+struct bt_conn_le_subrate_param {
+	/** Minimum subrate factor. */
+	uint16_t subrate_min;
+	/** Maximum subrate factor. */
+	uint16_t subrate_max;
+	/** Maximum Peripheral latency in units of subrated connection intervals. */
+	uint16_t max_latency;
+	/** Minimum number of underlying connection events to remain active
+	 *  after a packet containing a Link Layer PDU with a non-zero Length
+	 *  field is sent or received.
+	 */
+	uint16_t continuation_number;
+	/** Connection Supervision timeout (N * 10 MS).
+	 *  If using @ref bt_conn_le_subrate_set_defaults, this is the
+	 *  maximum supervision timeout allowed in requests by a peripheral.
+	 */
+	uint16_t supervision_timeout;
+};
+
+/** Subrating information for LE connections */
+struct bt_conn_le_subrating_info {
+	/** Connection subrate factor. */
+	uint16_t factor;
+	/** Number of underlying connection events to remain active after
+	 *  a packet containing a Link Layer PDU with a non-zero Length
+	 *  field is sent or received.
+	 */
+	uint16_t continuation_number;
+};
+
+/** Initialize subrating parameters
+ *
+ * @param  __subrate_min Minimum subrate factor.
+ * @param  __subrate_max Maximum subrate factor.
+ * @param  ___max_latency Maximum Peripheral latency in units of subrated
+ *                        connection intervals.
+ * @param  ___continuation_number Minimum number of underlying connection events
+ *                                to remain active after a packet containing a
+ *                                Link Layer PDU with a non-zero Length
+ *                                field is sent or received.
+ * @param  ___supervision_timeout Connection Supervision timeout (N * 10 MS).
+ */
+#define BT_CONN_LE_SUBRATE_PARAM_INIT(_subrate_min, _subrate_max, _max_latency, \
```

Comment:
Are these macros needed or should we require the application to create the struct manually like you do in the shell? A common complaint about zephyr and the bluetooth api is the heavy use of macros.

Reply:
My preference is for less macros as well, I added them to keep consistency with conn. update, PHY update and DLE in the same header, but happy to remove.